JSDemos/Demos/TreeList/LocalReordering/Vue/App.vue

```diff
@@ -91,28 +91,29 @@ export default {
       }
     },
     onReorder(e) {
-      let visibleRows = e.component.getVisibleRows(),
-        sourceData = e.itemData,
-        targetData = visibleRows[e.toIndex].node.data,
-        employees = [...this.employees];
+      const visibleRows = e.component.getVisibleRows();
 
       if (e.dropInsideItem) {
-        e.itemData.Head_ID = targetData.ID;
+        e.itemData.Head_ID = visibleRows[e.toIndex].key;
+
         e.component.refresh();
       } else {
-        let sourceIndex = employees.indexOf(sourceData),
-          targetIndex = employees.indexOf(targetData);
+        const sourceData = e.itemData;
+        const toIndex = e.fromIndex > e.toIndex ? e.toIndex - 1 : e.toIndex;
+        let targetData = toIndex >= 0 ? visibleRows[toIndex].node.data : null;
 
-        if (sourceData.Head_ID !== targetData.Head_ID) {
-          sourceData.Head_ID = targetData.Head_ID;
-          if (e.toIndex > e.fromIndex) {
-            targetIndex++;
-          }
+        if (targetData && e.component.isRowExpanded(targetData.ID)) {
+          sourceData.Head_ID = targetData.ID;
+          targetData = null;
+        } else {
+          sourceData.Head_ID = targetData ? targetData.Head_ID : -1;
         }
 
-        employees.splice(sourceIndex, 1);
-        employees.splice(targetIndex, 0, sourceData);
-        this.employees = employees;
+        var sourceIndex = this.employees.indexOf(sourceData);
+        this.employees.splice(sourceIndex, 1);
+
+        var targetIndex = this.employees.indexOf(targetData) + 1;
+        this.employees.splice(targetIndex, 0, sourceData);
```

Comment:
Use const instead of var
```suggestion
        const sourceIndex = this.employees.indexOf(sourceData);
        this.employees.splice(sourceIndex, 1);

        const targetIndex = this.employees.indexOf(targetData) + 1;
        this.employees.splice(targetIndex, 0, sourceData);
```